tasks/config.yml

```diff
@@ -1,12 +1,24 @@
 ---
 - name: Drop any groovy scripts
   template:
-    src: "{{ item }}"
+    src: "{{ item }}.groovy"
     dest: "{{ jenkins_home }}/init.groovy.d/"
-  with_fileglob:
-    - templates/*groovy
+  when: item != 'skip'
+  register: result_groovy
+  with_items:
+    - "{{ 'github_folder' if jenkins_deploy_gh_folder else 'skip' }}"
+    - general_config
+    - creds
+    - "{{ 'ec2_cloud' if jenkins_deploy_ec2_clouds else 'skip' }}"
+    - "{{ 'github_auth' if jenkins_deploy_gh_oauth else 'skip' }}"
```

Comment:
This looks like enough rope to hang yourself with, but I see the value in providing overrides.